lib/menu.rb

```diff
@@ -0,0 +1,36 @@
+class Menu
+
+  attr_reader :name
+
+  def initialize(name, *dishes)
+    @name = name
+    @dishes = Hash.new
+    load_dishes(dishes)
+  end
+
+  def dishes
+    @dishes.dup
+  end
+
+  def to_s
+    string = "#{name} Menu\n"
+    @dishes.each do |dish, price|
+      string += "#{dish}".ljust(25)
+      string += "#{'%.02f' % price}".rjust(10)
+      string += "\n"
+    end
+    string
+  end
+
+  private
+
+  def load_dishes(dishes)
+    fail "No dishes in the menu" if dishes.empty?
+    dishes.each_slice(2) do |name, price|
+      fail "Invalid dish name '#{name}'" unless name.is_a?(String)
```

Comment:
Okay, understood. The reason I added type checks for menu instantiation alone was so that incorrect data couldn't get into the menu. One could very easily miss a comma, or enter items in the wrong order while inputting a list of dish name, dish price pairs separated by a comma.